service/internal/auth/casbin.go

```diff
@@ -111,15 +111,23 @@ func NewCasbinEnforcer(c CasbinConfig, logger *logger.Logger) (*Enforcer, error)
 	if err != nil {
 		return nil, fmt.Errorf("failed to create casbin enforcer: %w", err)
 	}
-
-	return &Enforcer{
+	enforcer := &Enforcer{
 		Enforcer:        e,
 		Config:          c,
 		Policy:          c.Csv,
 		isDefaultPolicy: isDefaultPolicy,
 		isDefaultModel:  isDefaultModel,
 		logger:          logger,
-	}, nil
+	}
+
+	// If using SQL policy storage (opt-in), seed the store if empty
+	if c.EnableSQL {
+		if err := enforcer.useSQLPolicy(m); err != nil {
+			return nil, err
+		}
+	}
+
```

Comment:
If we change this to `Adapter` and then use a switch with `csv` as default. This way we can leave flexibility without flag backwards compatibility.